src/server/transaction.cc

```diff
@@ -1066,6 +1066,7 @@ bool Transaction::ScheduleUniqueShard(EngineShard* shard) {
   // Fast path. If none of the keys are locked, we can run briefly atomically on the thread
   // without acquiring them at all.
   if (quick_run) {
+    // TBD add acquire lock here
```

Comment:
what exactly do you mean by that? acquire of what?

Reply:
shard->db_slice().Acquire(mode, lock_args);